### PR TITLE
Feat/co applicant modal

### PR DESCRIPTION
**Pull request by kajm**
## Explain the changes you’ve made
Added the ability to add a co applicant when opening a new application

## Explain why these changes are made
A user needs to be able to add a co applicant when applying for a new application, and to make our work easier, we want the co applicant to be added on a case before the case is opened, hence these changes are made.

## Explain your solution
Added two new modals, one for opening a form and/or opening the add co applicant modal, and the addCoApplicant modal for entering a personal number for the co applicant.

Added API for adding a new co applicant.

## How to test

1. Checkout this branch
2. Fire upp the simulator by running the command `yarn ios` or `yarn android`
3. Make sure you have a new application case in backend
4. Start a new application

## Tested environments

- [x] Storybook on a iOS device/simulator.
- [] Building the Application on a iOS device/simulator.
- [] Building the Application on a Android device/simulator.

## Screenshots
simulator_screenshot_512D43D7-EA83-45B4-8553-B63B170DA7E6

simulator_screenshot_C4D15653-126B-47FA-A240-A6B5C4F43B4F

### CODE REVIEW

**Comment by helsingborgsstad** [comment]
Task linked: [CU-2q9vhfg Skapa ny modal med input-funktion i appen](https://app.clickup.com/t/2q9vhfg)

**Review by MasterKenth — CHANGES_REQUESTED** [review]
(I haven't tested that pin/other modals still work.)

File: source/components/atoms/TextButton/TextButton.tsx [hunk 1]
```diff
@@ -0,0 +1,20 @@
+import React from "react";
+
+import { TextButtonContainer, ButtonText } from "./TextButton.styled";
+
+interface TextButtonProps {
```
[the remark this answers is not in the document]
Comment (kajm) [reply source/components/atoms/TextButton/TextButton.tsx:5]: Do we need to export something that isn't used by anything else?

File: source/components/organisms/AddCoApplicantModal/AddCoApplicantModal.tsx [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+import { Modal } from "../../molecules/Modal";
+
+import { BackgroundBlurWrapper } from "../../atoms/BackgroundBlur";
+import { Text, Button, Input, TextButton } from "../../atoms";
+
+import { ValidationHelper } from "../../../helpers";
+
+import { DialogContainer, Container } from "./AddCoApplicantModal.styled";
+
+interface AddCoApplicantContentProps {
```
[the remark this answers is not in the document]
Comment (kajm) [reply source/components/organisms/AddCoApplicantModal/AddCoApplicantModal.tsx:14]: Same question as above

File: source/components/organisms/StartNewApplicationModal/StartNewApplicationModal.tsx [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+  StyledButton,
+} from "./StartNewApplicationModal.styled";
+
+interface StartNewApplicationModalProps {
+  visible: boolean;
+  onClose: () => void;
+  onOpenForm: () => void;
+  onChangeModal: () => void;
+}
+export default function StartNewApplicationModal({
```
[the remark this answers is not in the document]
Comment (kajm) [reply source/components/organisms/StartNewApplicationModal/StartNewApplicationModal.tsx:21]: I dont have a better name than the one you suggested. Lets start with that one

**Comment by kajm** [comment]
> 



> (I haven't tested that pin/other modals still work.)

Me neither. I need help testing that before this PR can be merged

**Comment by kajm** [comment]
> 



> (I haven't tested that pin/other modals still work.)

Me neither. I need help testing that before this PR can be merged

**Review by dannil76 — APPROVED** [review]
It works

**Review by MasterKenth — CHANGES_REQUESTED** [review]

File: source/components/organisms/AddCoApplicantModal/AddCoApplicantModal.tsx [hunk 4]
```diff
@@ -5,38 +5,85 @@ import Wrapper from "../../molecules/Dialog/Wrapper";
...
+    },
+  ];
+
+  const invalidInput =
+    inputValue.personalNumber.length !== 12 ||
+    !inputValue.firstName ||
+    !inputValue.lastName;
+
   return (
     <Modal
```
[the remark this answers is not in the document]
Comment (kajm) [reply source/components/organisms/AddCoApplicantModal/AddCoApplicantModal.tsx:88]: Not actually using it. But will add it so the warning will disappear.

**Pull request merged**